packages/flutter/lib/src/gestures/scale.dart

```diff
@@ -410,12 +414,27 @@ class ScaleGestureRecognizer extends OneSequenceGestureRecognizer {
   void _update() {
     final int count = _pointerLocations.keys.length;
 
+    final Offset? previousFocalPoint = _updated ? _currentFocalPoint : null;
```

Comment:
nit: instead of using an extra boolean to store `_update`, is it possible to make `_currentFocalPoint` an `Offset?` ?

Reply:
Yes if I add a few exclamation points where it's used, good call.